File: packages/php-sdk/src/Types/Requests/EmbeddedRecipientFields.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Requests;

/**
 * Shorthand field placement by text anchor for embedded signing. Each value is the anchor text to
 * replace (placement 'replace' + a default size). Mirrors the TypeScript SDK's
 * `EmbeddedRecipientFields`.
 */
final class EmbeddedRecipientFields
{
    /**
     * @param string|null $signature Anchor text for the signature field (e.g. '{signature1}')
     * @param string|null $date Anchor text for the date field
     * @param string|null $initials Anchor text for the initials field
     * @param string|null $fullName Anchor text for the full-name field
     */
    public function __construct(
        public ?string $signature = null,
        public ?string $date = null,
        public ?string $initials = null,
        public ?string $fullName = null,
    ) {}
}

File: packages/java-sdk/src/main/java/com/turbodocx/models/EmbeddedRecipientFields.java
package com.turbodocx.models;

/**
 * Shorthand field placement by text anchor; expands to full {@link Field} objects
 * (placement {@code replace} + a default size). Each value is the anchor text to replace,
 * e.g. {@code "{signature1}"}. A null key is skipped.
 */
public class EmbeddedRecipientFields {
    private final String signature;
    private final String date;
    private final String initials;
    private final String fullName;

    private EmbeddedRecipientFields(Builder builder) {
        this.signature = builder.signature;
        this.date = builder.date;
        this.initials = builder.initials;
        this.fullName = builder.fullName;
    }

    public String getSignature() {
        return signature;
    }

    public String getDate() {
        return date;
    }

    public String getInitials() {
        return initials;
    }

    public String getFullName() {
        return fullName;
    }

    public static class Builder {
        private String signature;
        private String date;
        private String initials;
        private String fullName;

        public Builder signature(String anchor) {
            this.signature = anchor;
            return this;
        }

        public Builder date(String anchor) {
            this.date = anchor;
            return this;
        }

        public Builder initials(String anchor) {
            this.initials = anchor;
            return this;
        }

        public Builder fullName(String anchor) {
            this.fullName = anchor;
            return this;
        }

        public EmbeddedRecipientFields build() {
            return new EmbeddedRecipientFields(this);
        }
    }
}
